src/pydicom/pixel_data_handlers/gdcm_handler.py

```diff
@@ -114,16 +114,15 @@ def create_data_element(ds: "Dataset") -> "DataElement":
     tsyntax = ds.file_meta.TransferSyntaxUID
     data_element = gdcm.DataElement(gdcm.Tag(0x7FE0, 0x0010))
     if tsyntax.is_compressed:
-        if get_nr_frames(ds, warn=False) > 1:
-            pixel_data_sequence = pydicom.encaps.decode_data_sequence(ds.PixelData)
-        else:
-            pixel_data_sequence = [pydicom.encaps.defragment_data(ds.PixelData)]
-
         fragments = gdcm.SequenceOfFragments.New()
-        for pixel_data in pixel_data_sequence:
-            fragment = gdcm.Fragment()
-            fragment.SetByteStringValue(pixel_data)
-            fragments.AddFragment(fragment)
+        nr_frames = get_nr_frames(ds, warn=False)
+        func = generate_fragmented_frames(ds.PixelData, number_of_frames=nr_frames)
```

Comment:
'func'?

Reply:
You don't like the variable name? Or because it's a generator rather than a function? I've changed it to something a bit more descriptive, but I figured since it gets used immediately something short and generic was OK.

Reply:
Yeah, it doesn't really matter, it just made me stop and think if I hadn't understood something...